### PR TITLE
Make downloading parent pom's optional to reduce false negatives

### PR DESCRIPTION
**Pull request by anuccio1**
## Description

A ton of maven projects try to download parent POM's externally, and were failing, causing FOSSA to show "0 Dependencies". This PR makes this step optional, allowing srclib-java to still pick up dependencies if an error is thrown trying to download parent POM files.

## Changes

All new files are vendored from: https://github.com/apache/maven/tree/maven-3.2.3

In addition to copying vendored code, we also make the following changes:

### Change 1:

https://github.com/apache/maven/blob/33f8c3e1027c3ddde99d3cdebad2656a31e8fdf4/maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java#L309

We now wrap the `readParent` call in a try/catch so that it doesn't completely fail the toolchain (see `src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java`: 

https://github.com/fossas/srclib-java/compare/master...fossas:optional-parent-pom?expand=1#diff-7eaf4fded50780c1825ca48bf3e4873213c4acc04534b6d8350ce73085bc83a6R321)

### Change 2:

https://github.com/apache/maven/blob/maven-3.2.3/maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java#L925

The `FATAL` is changed to a `WARNING` so it doesn't throw (see `src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java`: https://github.com/fossas/srclib-java/compare/master...fossas:optional-parent-pom?expand=1#diff-7eaf4fded50780c1825ca48bf3e4873213c4acc04534b6d8350ce73085bc83a6R938)

## Testing

To test, you should first try importing a project locally on the `develop` branch in fossa core. It should show 
You can fork and import this as a good example: https://github.com/anuccio1/nanohttpd.

This should show up with 0 dependencies.

After that is done, point your image at `quay.io/fossa/fossa:srclibjava-parentpom-docker-build-28fdf7dd`, and try analyzing a different commit.

This should show up with ~56 dependencies (non-zero).

## Notes
For previous Build failure logs, you can view this build log:

(located here: https://app.fossa.com/projects/git%2Bgithub.com%2Fgp-fossa%2Fnanohttpd/refs/branch/master/efb2ebf85a2b06f7c508aba9eaad5377e3a01e81/browse/dependencies)

Org: FOSSA
Project: NanoHttpd

GH link: https://github.com/anuccio1/nanohttpd
[build-5577658.log](https://github.com/fossas/srclib-java/files/5776791/build-5577658.1.log)

### CODE REVIEW

File: src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java [hunk 1]
```diff
@@ -0,0 +1,1226 @@
...
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Benjamin Bentmann
+ */
+@Component( role = ModelBuilder.class )
+public class DefaultModelBuilder
+    implements ModelBuilder
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultModelBuilder.class);
```
Comment (anuccio1) [inline src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java:82]: change here

File: src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java [hunk 2]
```diff
@@ -0,0 +1,1226 @@
...
+            configureResolver( request.getModelResolver(), tmpModel, problems );
+
+            ModelData parentData;
+            try {
+                parentData = readParent( tmpModel, currentData.getSource(), request, problems );
+            } catch (Exception e) {
+                LOGGER.warn("Could not read parent POM", e);
+                parentData = null;
+                LOGGER.warn("Continuing without parent POM...");
+            }
```
Comment (anuccio1) [inline src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java:322]: changes here to wrap in try/catch

File: src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java [hunk 3]
```diff
@@ -0,0 +1,1226 @@
...
+            if ( childModel != problems.getRootModel() )
+            {
+                buffer.append( " of POM " ).append( ModelProblemUtils.toSourceHint( childModel ) );
+            }
+            buffer.append( " points at " ).append( groupId ).append( ":" ).append( artifactId );
+            buffer.append( " instead of " ).append( parent.getGroupId() ).append( ":" ).append( parent.getArtifactId() );
+            buffer.append( ", please verify your project structure" );
+
+            problems.setSource( childModel );
+            problems.add( new ModelProblemCollectorRequest( Severity.WARNING, Version.BASE )
```
Comment (anuccio1) [inline src/main/java/com/sourcegraph/javagraph/maven/vendored/DefaultModelBuilder.java:855]: change here from `FATAL` to `WARNING`

File: src/main/java/com/sourcegraph/javagraph/MavenProject.java [hunk 4]
```diff
@@ -1,6 +1,8 @@
 package com.sourcegraph.javagraph;
 
 import com.google.common.collect.Iterators;
+import com.sourcegraph.javagraph.maven.vendored.*;
+import com.sourcegraph.javagraph.maven.vendored.DefaultModelBuilderFactory;
```
Comment (anuccio1) [inline src/main/java/com/sourcegraph/javagraph/MavenProject.java:4]: we will use the vendored `DefaultModelBuilderFactory` in favor of the apache one

File: .gitignore [hunk 5]
```diff
@@ -18,3 +18,4 @@ dependency-reduced-pom.xml
 
 /bundled-jdk/
 .vscode/
+.java-version
```
Comment (Serubin) [inline .gitignore:21]: [nit] newline

**Review by Serubin — APPROVED** [review]
Looks good - testing locally resulted in 9 deps on this branch over 0 on develop using nanohttp.

If you expected more, we can pair and look into it.

**Pull request merged**